server/Dockerfile

```diff
@@ -1,23 +1,24 @@
-# This stage is tagged for local development.
+FROM curlimages/curl AS download
+WORKDIR /var/tmp
+RUN curl -O https://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver \
+    -O https://hgdownload.cse.ucsc.edu/goldenpath/hg19/liftOver/hg19ToHg38.over.chain.gz \
+    -O https://hgdownload.cse.ucsc.edu/goldenpath/hg38/liftOver/hg38ToHg19.over.chain.gz && \
+    gunzip ./hg38ToHg19.over.chain.gz && \
+    gunzip ./hg19ToHg38.over.chain.gz
+
 FROM node:16-slim AS dev
 WORKDIR /usr/src/app
-RUN apt-get update && apt-get install -y wget libcurl4
-RUN wget -O /usr/local/bin/liftOver https://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver  \
-  && chmod +x /usr/local/bin/liftOver
-COPY . .
-WORKDIR /home/node
-RUN mkdir chains && \
-  wget -O ./chains/hg19ToHg38.over.chain.gz https://hgdownload.cse.ucsc.edu/goldenpath/hg19/liftOver/hg19ToHg38.over.chain.gz && \
-  wget -O ./chains/hg38ToHg19.over.chain.gz https://hgdownload.cse.ucsc.edu/goldenpath/hg38/liftOver/hg38ToHg19.over.chain.gz && \
-  gunzip ./chains/hg38ToHg19.over.chain.gz && \
-  gunzip ./chains/hg19ToHg38.over.chain.gz
-RUN chown -R node:node ./ && chown -R node:node /home/node
-WORKDIR /usr/src/app
+COPY --from=download /var/tmp/liftOver /usr/local/bin 
+RUN chmod +x /usr/local/bin/liftOver && apt-get update && apt-get install -y libcurl4
+COPY --from=download --chown=node:node /var/tmp/*.over.chain /home/node/
 
 # This layer is shared between the builder and production stages
-FROM dev AS base
+FROM node:16-slim AS base
 WORKDIR /usr/src/app
 COPY package*.json yarn.lock ./
+COPY --from=download /var/tmp/liftOver /usr/local/bin
+RUN chmod +x /usr/local/bin/liftOver && apt-get update && apt-get install -y libcurl4
```

Comment:
`--chmod` is also an option to the copy directive, and you shouldn't need libcurl here anymore then?

Reply:
[https://stackoverflow.com/questions/67910547/why-docker-copy-doesnt-change-file-permissions-chmod](https://stackoverflow.com/questions/67910547/why-docker-copy-doesnt-change-file-permissions-chmod) 

`--chmod` option for `COPY` requires BUILDKIT, should we add `--chmod` and build with `DOCKER_BUILDKIT=1 docker-compose ........` ?